File: src/app/types/prediction/BetPredictionContext.ts
import LeagueStats from '../stats/LeagueStats';
import MatchStats from '../stats/MatchStats';
import TeamStats from '../stats/TeamStats';
import { BetTypeDefinition } from './Bet';

export default interface BetPredictionContext {
  match: MatchStats;
  teamStats?: TeamStats;
  leagueStats?: LeagueStats;
  bet: BetTypeDefinition;
}

File: footystats-tools-backend/src/main/java/de/ludwig/footystats/tools/backend/services/prediction/influencer/BetPredictionContext.java
package de.ludwig.footystats.tools.backend.services.prediction.influencer;

import de.ludwig.footystats.tools.backend.services.stats.LeagueStats;
import de.ludwig.footystats.tools.backend.services.stats.MatchStats;
import de.ludwig.footystats.tools.backend.services.stats.TeamStats;
import de.ludwig.footystats.tools.backend.services.prediction.Bet;

import java.util.List;

public record BetPredictionContext(MatchStats match,
                                   List<TeamStats> teamStats,
                                   LeagueStats leagueStats,
                                   Bet bet) {
}
